src/util/profile/prof_file.c

```diff
@@ -368,7 +368,8 @@ errcode_t profile_update_file_data_locked(prf_data_t data, char **ret_modspec)
         retval = profile_process_directory(data->filespec, &data->root);
     } else {
         retval = profile_parse_file(f, &data->root, ret_modspec);
-        (void)fclose(f);
+        if (f != NULL)
```

Comment:
This condition is always true.  Is there actually a compiler warning if you omit it, or were you just adding it to be defensive?